Add sdl2_image library

patrigg opened the pull request:
SDL2_image adds extended image loading capabilities to SDL2 allowing many more file formats to be loaded into an SDL_Surface, like PNG, JPEG, TIFF, WEBP,...

This port enables all file formats which SDL2_image can handle internally. PNG, JPEG, TIFF and WEBP file formats are supported through external libraries. This port will automatically link to all installed libraries at the time SDL2_image is installed.
If a library is not available the port will fall back to linking at run time. In this case, dlls have to be copied manually to the application directory since vcpkg cannot resolve them.

I used the FindWEBP.cmake file from the third party WebKit library for resolving the dependency to WebP.
Submission containing materials of a third party:
FindWEBP.cmake by Raphael Kubo da Costa <rakuco@webkit.org> and Igalia S.L. taken from WebKit: https://github.com/WebKit/webkit/blob/master/Source/cmake/FindWebP.cmake

patrigg committed:
updated zlib port to version 1.2.10

patrigg committed:
Merge remote-tracking branch 'upstream/master'

patrigg committed:
[zlib] fixed wrong FILENAME

sdcb committed:
Add dx port.

codicodi committed:
[freetype] specify dependencies

patrigg committed:
Merge branch 'master' of https://github.com/Microsoft/vcpkg

patrigg committed:
added first version of SDL2_image library which currently does not link to any external libraries.

patrigg committed:
forces CMake use zlib version which is installed by vcpkg.

patrigg committed:
Merge branch 'master' of https://github.com/patrigg/vcpkg

patrigg committed:
Merge branch 'sdl2_image' of https://github.com/patrigg/vcpkg

patrigg committed:
Merge branch 'master' of https://github.com/patrigg/vcpkg

patrigg committed:
Revert "forces CMake use zlib version which is installed by vcpkg."

This reverts commit d62dda3b030e31f322aaa47b4aeaa0a12fa3201b.

codicodi committed:
add libarchive

traversaro committed:
vcpkg_apply_patches: add QUIET option

traversaro committed:
cmake: add qmake-related helpers function

traversaro committed:
Add qwt port

patrigg committed:
finished port of sdl2_image
 - optional dependencies which are installed in vcpkg are automatically linked
 - unavailable dependencies are marked as dynamically linked and dlls have to be copied manually if respective file formats are used

patrigg committed:
Merge branch 'master' of https://github.com/Microsoft/vcpkg

patrigg committed:
Merge branch 'sdl2_image'

alexkaratarakis committed:
Merge pull request #491 from codicodi/fix-ft-deps

[freetype] specify dependencies

alexkaratarakis committed:
Merge pull request #490 from sdcb/dx

Add dx port.

codicodi commented on ports/sdl2-image/FindWEBP.cmake:
```diff
@@ -0,0 +1,51 @@
+# Look for the library.
+find_library(
+    WEBP_LIBRARIES
+    NAMES webp
```
Will this find debug version? It is named `webpd`.

patrigg replied:
Thanks for reviewing. No, it won't find the debug version.

patrigg committed:
fixed findWEBP.cmake to select debug version of webp for debug builds.

patrigg commented:
The issue with finding the debug version is now fixed.

alexkaratarakis committed:
Merge pull request #493 from codicodi/add-archive

add libarchive

traversaro commented:
For reference, I wrote a [`FindWEBP.cmake` file](https://github.com/Microsoft/vcpkg/blob/master/ports/freeimage/cmake/FindWEBP.cmake) while writing the `freeimage` port. I don't think it is a big problem to have two different `FindWEBP.cmake` files in different ports, I wanted just to point out to the already existing one (that has no copyright attribution problems, as I wrote it ex-novo).

alexkaratarakis committed:
Merge pull request #494 from traversaro/add-qwt

Add qwt port

traversaro commented:
> This port enables all file formats which SDL2_image can handle internally. PNG, JPEG, TIFF and WEBP file formats are supported through external libraries. This port will automatically link to all installed libraries at the time SDL2_image is installed.
If a library is not available the port will fall back to linking at run time. In this case, dlls have to be copied manually to the application directory since vcpkg cannot resolve them.

I don't think it is a good idea to have a port that change what is building depending on the state of the vcpkg installation, as it would be extremely difficult to debug all possible combination of available/not available ports (see for example https://github.com/Microsoft/vcpkg/issues/412 ).
I think it is cleaner to just add `libpng, libjpeg-turbo, tiff, libwebp` to `Build-Depends:` to be sure that all dependencies are always available, and always link them in the "classical" way.

traversaro committed:
[qt5] Avoid printing warnings for problems in applying"confguration patches"

The `QUIET` option for `vcpkg_apply_patches` was added in https://github.com/Microsoft/vcpkg/commit/f4c34bb42dd0fe8217d4d54a9a4a2eeecdb0f0f4 . 
Fixes https://github.com/Microsoft/vcpkg/issues/411 .

traversaro committed:
[qt5] Fix tabs/spaces mismatch

patrigg committed:
[sdl2-image] added dependencies to CONTROL so they will always be statically linked
Replaced FindWEBP.cmake with version from freeimage port

patrigg commented:
I switched to your version of FindWEB.cmake. However, why are you setting WEBP_LIBRARIES to WEBP_LIBRARY? Both variables should be set correctly by select_library_configurations.

Though I think the current approach with linking is more flexible, I get your point regarding issues with bugs based on port availability. I added the build dependencies as you suggested.

alexkaratarakis committed:
[vcpkg_apply_pathces] Fix typo (also trailing whitespace)

traversaro commented:
I think you are right about setting WEB_LIBRARY and setting WEBP_LIBRARIES to WEBP_LIBRARY , those two lines can be probably be removed.

alexkaratarakis committed:
[libvorbis] Add version

alexkaratarakis committed:
[spriv-tools] Add version and description

codicodi commented:
I'd also like to point out that installation dir for the header file seems quite nonstandard - most of the other distributions put it in include/SDL2.

patrigg committed:
[sdl2-image] changed include directory to SDL2 include directory

patrigg commented:
@codicodi changed the installation dir as you suggested

alexkaratarakis committed:
Merge pull request #497 from traversaro/patch-3

[qt5] Avoid printing warnings for problems in applying"confguration patches"

patrigg committed:
Merge branch 'master' of https://github.com/patrigg/vcpkg

patrigg commented:
I squashed my commits. Is there anything I need to do to?

ras0219-msft commented:
Awesome library, thanks for the port. GitHub was not diffing correctly so it took a bit longer to review :).
